### DISCUSSION TITLE
Screen functionality?

### DISCUSSION THREAD

**Discussion by Nuuki9** [ajfriesen/ESPHome-AirGradient#3] [Q&A] [answered]
This looks good - I already have some AirGradient PCBs and the sensors are on the way. I'm planning to add a TVOC sensor as well so hopefully won't mess that up.

One question - am I right to assume that using ESPHome rather than the original arduino firmware means the screen won't function? Its not a big deal if not but if so I'd build it without.

Thanks again!

**Comment by ajfriesen** [comment, marked as answer]
The screen will work very well my fellow nerd!

You can see that screen in action on my blog post about this build:
[Measuring Air Quality in your Home Office](https://www.ajfriesen.com/measuring-air-quality-in-your-home-office/
)

You can see the code for the display here:
https://github.com/ajfriesen/ESPHome-AirGradient/blob/main/air-gradient.yaml#L44

I think it should even be possible to display any data from home assistant on that screen. Like your smartphone percentage or whatever you want.

**Reply by Nuuki9** [reply]
Perfect - thanks for the link to your great post. Sounds like I'll be able to include info from the VOC sensor I want to add as well, so that's even better :-)

**Reply by ajfriesen** [reply]
I thought I had linked that blog post to the README.md, but man I did not!
Should do that for people who find this via GitHub or Google directly and not via my blog post!

**Comment by Impatience71** [comment]
Just one idea which comes to my mind: some years before I was also working with the BME680. Would do you agree that's worth a try to try out the BME680 in the air gradient? I know about the issue with the "not open" Bosch libraries.

**Reply by Nuuki9** [reply]
> Just one idea which comes to my mind: some years before I was also working with the BME680. Would do you agree that's worth a try to try out the BME680 in the air gradient? I know about the issue with the "not open" Bosch libraries.

The only thing you'd gain from a BME680 would be gas, though I'm not sure if its actually measuring VOC - it sounds like an "equivalent" value. I'm planning to add an SGP30 to handle VOC.

**Reply by ajfriesen** [reply]
Sorry, can not help on that topic, no experience with the BME680.
